scripts/build/build/target.py

```diff
@@ -264,6 +285,50 @@ def HumanString(self):
 
         return result
 
+    def ToDict(self):
+        """Outputs a parseable description of the available variants
+        and modifiers:
+
+            like:
+
+            {
+                "name": "foo"
+                "shorthand": "foo-bar-baz[-m1]
+                "parts": [
+                    {
+                        "name": "foo",
+                        "build_arguments": {
+                            "board": "bar"
+                        }
+                    }
+                    {
+                        "name": "baz",
+                        "build_arguments": {
+                            "app": "foo.baz"
+                        }
+                    }
+                ],
+                "modifiers": [
+                    {
+                        "name": "modifier1",
+                        "m1": "True"
+                    }
+                ]
+            }
+        """
+        result: Dict[str, Any] = {}
+
+        result['name'] = self.name
+        result['shorthand'] = self.HumanString()
+
+        result['parts']: List[str] = []
+        for target in self.fixed_targets:
+            result['parts'] = [part.ToDict() for part in target]
```

Comment:
Aren't you overwriting the same dict key over and over again?

Also, whey the dict is created key-by-key instead of e.g.:
```python
return {
 'name': self.name,
 'shorthand': self.HumanString(),
  # I'm not sure whether such output is intended...
 'parts': [part.ToDict() for target in self.fixed_targets for part in target], 
 'modifiers': [part.ToDict() for part in self.modifiers],
}
```

Reply:
Yes, it is. Since most parts only had 1 item, the output looked ok. Parts like this were incorrect:
`TargetPart('BL602-IoT-Matter-V1', board=BouffalolabBoard.BL602_IoT_Matter_V1, module_type="BL602"),`